extensions/vertx-http/runtime/src/main/java/io/quarkus/devui/runtime/MvnpmHandler.java

```diff
@@ -37,20 +44,38 @@ public void handle(RoutingContext event) {
         }
 
         try {
-            InputStream is = mvnpmLoader.getResourceAsStream(BASE_DIR + fullPath);
-            if (is != null) {
-                byte[] contents = is.readAllBytes();
-                event.response()
-                        .putHeader(HttpHeaders.CONTENT_TYPE, getContentType(fileName))
-                        .end(Buffer.buffer(contents));
-                return;
+            URL url = mvnpmLoader.getResource(BASE_DIR + fullPath);
+            URLConnection openConnection = url.openConnection();
+            long lastModified = openConnection.getLastModified();
+            try (InputStream is = openConnection.getInputStream()) {
+                if (is != null) {
+                    byte[] contents = is.readAllBytes();
+                    event.response()
+                            .putHeader(HttpHeaders.CONTENT_TYPE, getContentType(fileName))
+                            .putHeader(HttpHeaders.CACHE_CONTROL, "public, immutable, max-age=31536000")
+                            .putHeader(HttpHeaders.LAST_MODIFIED, formatDate(lastModified))
+                            .putHeader("date", formatDate(LocalDateTime.now()))
+                            .end(Buffer.buffer(contents));
+                    return;
+                }
             }
         } catch (IOException ex) {
             throw new UncheckedIOException(ex);
         }
         event.next();
     }
 
+    private String formatDate(long m) {
+        Instant i = Instant.ofEpochMilli(m);
+        return formatDate(i);
+    }
+
+    private String formatDate(TemporalAccessor t) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("EEE, dd MMM yyyy HH:mm:ss z", Locale.ENGLISH)
```

Comment:
Isn't that a bit expensive? 
Isn't that the API that was made thread-safe? 

@geoand do you know?

Reply:
> Isn't that the API that was made thread-safe

Correct. 

> Isn't that a bit expensive?

Yeah, it's not very performant AFAIK. Netty has `DateFormatter` but I don't know how it compares and if it has been optimized. @franz1981 do you know?

Reply:
If it's an hot path probably worth using a whole different pattern...
The easier thing is to save in a static final field the formatter, which is immutable and thread safe (please check the doc yourself, I am still on PTO so I have limited GitHub access..).

The best would be to have a volatile field with the formatted string, updated using a timer (or some scheduled task) with second granularity. It's less costly and in the hot path the best solution overall.

Reply:
As usual, measure, not guess; check if the date formatting is a costly hot path and you have few options to reduce its overhead...
@geoand I didn't checked Netty yes will do with my beloved desktop at hand 👍

Reply:
This is Dev mode only, and also Dev UI only, so for now I think we can merge.